java/org.travlyn.server/src/main/java/org/travlyn/server/service/TravlynService.java

```diff
@@ -477,11 +477,24 @@ public void updateTrip(Trip trip) throws NoResultException {
     @Transactional
     public ExecutionInfo getExecutionInfo(Long tripId, Long userId, double lat, double lon, boolean reorder, boolean isRoundtrip, String lang) throws NoResultException {
         Session session = sessionFactory.getCurrentSession();
-        reorder = false;
-        TripEntity trip = session.createQuery("from TripEntity where id = :id", TripEntity.class)
+        TripEntity tripEntity = session.createQuery("from TripEntity where id = :id", TripEntity.class)
                 .setParameter("id", toIntExact(tripId))
                 .getSingleResult();
-        OpenRouteTripDirectionRequest directionRequest = new OpenRouteTripDirectionRequest(lat, lon, trip.toDataTransferObject(), isRoundtrip, lang);
+
+        Trip trip = tripEntity.toDataTransferObject();
+        if (reorder) {
+            List<Stop> stops = trip.getStops();
+            if (isRoundtrip) {
+                // add stop that represents the position of the user
+                stops.add(0, new Stop().id(-1).latitude(lat).longitude(lon));
```

Comment:
Why do we add this stop to index 0? Shoulnd it be at the end of the Trip? I need some clarification here even after checking out and trying...